### ISSUE TITLE
job_tracker process doesn't finish and blocks subsequent invocations

### ISSUE THREAD

**Issue by bossie** [Open-EO/openeo-geopyspark-driver#134]
Occasionally the job_tracker processes on dataflow.intern.vgt.vito.be will not finish and subsequent invocations will pile up in the queue before it. This means that the status of all jobs for certain environment e.g. prod will not be updated.

The current "fix" is to wait for a user to complain and kill the corresponding (child of) the job_tracker process and empty the queue. A short-term improvement might be to send an alert if the queue reaches a certain level so there is less impact.

A better solution would be to find out why this process hangs and prevent it from doing so. It looks like the call to "yarn application -status" is the culprit:

```
nifi-in+ 19665  0.0  0.0   4300   568 ?        S    07:53   0:00          \_ scl enable rh-python38 -- bash job_tracker.sh
nifi-in+ 19667  0.0  0.0 113120  1396 ?        S    07:53   0:00          |   \_ /bin/bash /var/tmp/sclXG1UGw
nifi-in+ 19670  0.0  0.0 113128  1396 ?        S    07:53   0:00          |       \_ bash job_tracker.sh
nifi-in+ 19673  0.0  0.4 934048 131316 ?       Sl   07:53   0:03          |           \_ python -m openeogeotrellis.job_tracker
nifi-in+ 20068  0.0  0.7 2854100 229512 ?      Sl   07:54   0:10          |               \_ /usr/lib/jvm/java-openjdk/bin/java -Dproc_application -Dzookeeper.sasl.client=true -Dzookeeper.sasl.client.username=zookeeper -Djava.security.auth.login.config=/usr/hdp/current/hadoop-client/conf/yarn_jaas.conf -Dzookeeper.sasl.clientconfig=Client -Dzookeeper.sasl.client=true -Dzookeeper.sasl.client.username=zookeeper -Djava.security.auth.login.config=/usr/hdp/current/hadoop-client/conf/yarn_jaas.conf -Dzookeeper.sasl.clientconfig=Client -Dhdp.version=3.1.4.0-315 -Djava.net.preferIPv4Stack=true -Dhdp.version=3.1.4.0-315 -Dyarn.id.str= -Dyarn.policy.file=hadoop-policy.xml -Djava.io.tmpdir=/var/lib/ambari-server/data/tmp/hadoop_java_io_tmpdir -Dservice.libdir=/usr/hdp/current/hadoop-yarn-client/./,/usr/hdp/current/hadoop-yarn-client/lib,/usr/hdp/3.1.4.0-315/hadoop-hdfs/./,/usr/hdp/3.1.4.0-315/hadoop-hdfs/lib,/usr/hdp/3.1.4.0-315/hadoop/./,/usr/hdp/3.1.4.0-315/hadoop/lib -Xmx1024m -Dyarn.log.dir=/var/log/hadoop-yarn/nifi-intern -Dyarn.log.file=hadoop.log -Dyarn.home.dir=/usr/hdp/current/hadoop-yarn-client -Dyarn.root.logger=INFO,console -Djava.library.path=:/usr/hdp/3.1.4.0-315/hadoop/lib/native/Linux-amd64-64:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64:/var/lib/ambari-server/data/tmp/hadoop_java_io_tmpdir:/usr/hdp/3.1.4.0-315/hadoop/lib/native -Dhadoop.log.dir=/var/log/hadoop-yarn/nifi-intern -Dhadoop.log.file=hadoop.log -Dhadoop.home.dir=/usr/hdp/3.1.4.0-315/hadoop -Dhadoop.id.str=nifi-intern -Dhadoop.root.logger=INFO,console -Dhadoop.policy.file=hadoop-policy.xml -Dhadoop.security.logger=INFO,NullAppender org.apache.hadoop.yarn.client.cli.ApplicationCLI application -status application_1647270439955_62503
```

**Comment by bossie** [comment]
stderr output suggests a connection towards the RMs is endlessly retried.

[19665.stderr.txt](https://github.com/Open-EO/openeo-geopyspark-driver/files/8361770/19665.stderr.txt)

**Comment by bossie** [comment]
Fixed job_tracker script like async_task in #144.
